File: ALTableViewFrameworkExamples/ALTableViewFrameworkExamples/Master/MasterTableViewCell.m
//
//  MasterTableViewCell.m
//  ALTableViewFrameworkExamples
//
//  Created by lorenzo villarroel perez on 20/11/15.
//  Copyright © 2015 Abimael Barea Puyana. All rights reserved.
//

#import "MasterTableViewCell.h"

@implementation MasterTableViewCell

- (void)awakeFromNib {
    // Initialization code
}

- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
    [super setSelected:selected animated:animated];

    // Configure the view for the selected state
}

-(void) configureCell: (NSString *) object {
    self.label.text = object;
}

@end

File: ALTableViewSwift/ALTableView/Master/MasterTableViewCell.swift
//
//  MasterTableViewCell.swift
//  ALTableView
//
//  Created by lorenzo villarroel perez on 10/3/18.
//  Copyright © 2018 lorenzo villarroel perez. All rights reserved.
//

import UIKit

class MasterTableViewCell: UITableViewCell, ALCellProtocol {
    
    @IBOutlet weak var labelText: UILabel!
    
    override func awakeFromNib() {
        super.awakeFromNib()
        // Initialization code
    }

    override func setSelected(_ selected: Bool, animated: Bool) {
        super.setSelected(selected, animated: animated)

        // Configure the view for the selected state
    }
    
    public func cellCreated(dataObject: Any) {
        if let title = dataObject as? String {
            self.labelText.text = title
        }
    }
    
    public func cellPressed(viewController: UIViewController?) {
        print(self.labelText.text)
    }
    
//    func cellCreated(dataObject: Any) {
//        
//    }
    
}
